src/main/java/jenkins/plugins/logstash/configuration/ElasticSearch.java

```diff
@@ -163,5 +178,17 @@ public FormValidation doCheckUrl(@QueryParameter("value") String value)
       }
       return FormValidation.ok();
     }
+    public FormValidation doCheckMimeType(@QueryParameter("value") String value) {
+      if (StringUtils.isBlank(value)) {
```

Comment:
I think a blank mimeType should be valid. In that case it should be set to null in the DAO so you have the old behaviour. This should be mentioned in the help.
You can also save with a blank mimeType which would result in no mimeType being configured in the StringEntity. Not sure if this will break anything,

Reply:
@mwinter69  IMHO, I think we should not allow user to save config with blank mimeType and it would not be of much help/value addition if we do not configure mimeType on StringEntity considering the nature of elasticsearch and logstash. I think we should set application/json by default which will make both ES and logstash happy.

Reply:
Does this also fail when you try to save with an empty string? If not then you can run into problems at runtime as you will pass an empty string. Maybe use Util.fixEmptyAndTrim in the DAO to avoid this or do not allow to save with an illegal mimetype.

Reply:
@mwinter69 Apologies for delays in response. Recovering from bad health. Correct me if I mis-understand understand your question. As per current logic, user is not allowed to save an empty string for mimeType.

Reply:
Go to the configuration page and make the mimetype field empty. When you leave the field it will tell you that a value is required. Now click on apply. Congratulations you have set the mimetype to an empty string which will most likely fail.
You should throw an exception in the setter method if no valid mimetype is given.

Reply:
Thanks for elaborating @mwinter69. It was helpful.  Wasn't aware that it is allowed to submit a form even if there are validation errors. Learned something new today. Will make that change.

Reply:
@mwinter69 Need an input on implementation. As you suggested I am now throwing an exception from `ElasticSearch.setMimeType()` if user tries to save an invalid mimetype dispite of warnings on UI. The exception is thrown and invalid mimetype is not saved. However, UI does not display the thrown exception and it continues to display "Saved" banner. Can you please guide me on how to make my exception bubble up all the way to the UI?

Reply:
Looking into this it seems that Stapler is swallowing the exception and only logging when you try to use a setter method,
By initializing the mimetype in the constructor, and throwing the exception there you can avoid that it gets saved.

Reply:
Thanks for the inputs. I am hoping that I understood your suggestion correctly. Here is what I did. I initialized `mimeType` in `ElasticSearchDao()` constructor and if it is invalid, thrown an exception just like how its done for `uri.toURL();` currently. However, I realized that `ElasticSeachDao` is not instantiated every time we click on "Save" button on jenkins configuration page. Hence throwing exception in side `ElasticSearchDao()` did not work for me meaning UI does not show any exception. It only throws exception when you run a job. 

In an alternative approach and also you mentioned this in one of your earlier comments that we can allow blank `mimeType`. If the `mimeType` is blank we can set it to `null`  in `ElasticSearch` and with current implementation if `mimeType` is `null` it will be set to default `application/json` value. Does this sound acceptable?

Reply:
The dao is the wrong place to do the checks. You have to do it in the constructor of `configuration/ElasticSearch`. Alternatively one could extend the base class `LogstashIndexer` with a validate method and call the validate method in the `LogstashConfiguration.configure` before `save()`

Reply:
@mwinter69 Pushed updated code.

Reply:
@jakub-bochenski Updated code with your review comments